### PR TITLE
Create a database based on a diagram

### PR DESCRIPTION
**Pull request by carlylechia**
To complete this task, my partner and I:

- Created a new file called `schema_based_on_diagram.sql` and wrote all the code inside
- Built a new database called **clinic** according to the schema diagram in the [project requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/database_from_diagram.md)
- Respected the relationships and created a 'join table' for the many-to-many relationship between treatments and histories

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+CREATE TABLE medical_histories(id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY, admitted_at timestamp, patient_id INT, status VARCHAR(100), CONSTRAINT FK_patients_medhistories FOREIGN KEY(patient_id) REFERENCES patients(id));
+
+CREATE TABLE invoices(id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY, total_amount decimal, generated_at timestamp, payed_at timestamp, medical_history_id INT, CONSTRAINT FK_history_invoice FOREIGN KEY(medical_history_id) REFERENCES medical_histories(id), UNIQUE(medical_history_id));
+
+CREATE TABLE treatments(id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY, type VARCHAR(50), name VARCHAR(100));
+
+CREATE TABLE treatment_histories(id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY, history_id INT, treatment_id INT, FOREIGN KEY(history_id) REFERENCES medical_histories(id), FOREIGN KEY(treatment_id) REFERENCES treatments(id));
+
+CREATE TABLE invoice_items(id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY, unit_price decimal, quantity INT, total_price decimal, invoice_id INT, treatment_id INT, FOREIGN KEY(invoice_id) REFERENCES invoices(id), FOREIGN KEY(treatment_id) REFERENCES treatments(id));
```
Comment (keroloussamy) [inline schema_based_on_diagram.sql]: - [ ] Kindly add indexes for all your FKs as asked by the requirements `Remember to add the FK indexes.`.

**Review by keroloussamy — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Check the comments under the review._

- [ ] x

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (carlylechia) [reply schema_based_on_diagram.sql]: oh, thanks. we forgot

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @carlylechia @frank1738, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**